bitcoin/src/blockdata/script.rs

```diff
@@ -1222,6 +1236,66 @@ mod test {
         script = script.push_opcode(OP_CHECKSIG); comp.push(0xACu8); assert_eq!(&script[..], &comp[..]);
     }
 
+    #[test]
+    fn test_p2pk_pubkey_bytes() {
+        // happy path
+        let key_str = "0411db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3";
+        let key = PublicKey::from_str(key_str).unwrap();
+        let p2pk = Script::builder().push_key(&key).push_opcode(OP_CHECKSIG).into_script();
+        let actual = p2pk.p2pk_pubkey_bytes().unwrap();
+        assert_eq!(actual.to_vec(), key.to_bytes());
+
+        // invalid script checks
+        let no_checksig = Script::builder().push_key(&key).into_script();
+        assert_eq!(no_checksig.p2pk_pubkey_bytes(), None);
+        let empty_script = Script::builder().into_script();
+        assert!(empty_script.p2pk_pubkey_bytes().is_none());
+        let no_push_bytes = Script::builder().push_opcode(OP_CHECKSIG).into_script();
+        assert!(no_push_bytes.p2pk_pubkey_bytes().is_none());
+
+        // invalid key check
+        let malformed_key = "21032e58afe51f9ed8ad3cc7897f634d881fdbe49816429ded8156bebd2ffd1";
+        let invalid_p2pk_script = Script::builder().push_slice(malformed_key.as_bytes()).into_script();
+        assert!(invalid_p2pk_script.p2pk_pubkey_bytes().is_none());
```

Comment:
This test is not very useful since it doesn't contain `OP_CHECKSIG`. This function should return `Some` though.

Reply:
Why should it return `Some`, its an invalid script without the `OP_CHECKSIG`

Reply:
I meant it should have `OP_CHECKSIG` and when it has it it should return `Some`. Basically, it should return those bytes even when the key is invalid.

Reply:
Oh there was some confusion on this. The malformed_key is only 63 bytes long here. So it should return None